src/WebJobs.Script/OutOfProc/Http/HttpFunctionInvokeDispatcher.cs

```diff
@@ -0,0 +1,200 @@
+            _eventManager = eventManager;
+            _workerConfigs = languageWorkerOptions.Value.WorkerConfigs;
+            _logger = loggerFactory.CreateLogger<HttpFunctionInvokeDispatcher>();
+            _httpInvokerChannelFactory = httpInvokerChannelFactory;
+            _httpInvokerService = httpInvokerService;
+
+            State = FunctionDispatcherState.Default;
+
+            _workerErrorSubscription = _eventManager.OfType<HttpWorkerErrorEvent>()
+               .Subscribe(WorkerError);
+            _workerRestartSubscription = _eventManager.OfType<HttpWorkerRestartEvent>()
+               .Subscribe(WorkerRestart);
+        }
+
+        public FunctionDispatcherState State { get; private set; }
+
+        internal ConcurrentStack<HttpWorkerErrorEvent> LanguageWorkerErrors => _languageWorkerErrors;
+
+        internal async void InitializeJobhostLanguageWorkerChannelAsync()
+        {
+            await InitializeJobhostLanguageWorkerChannelAsync(0);
+        }
+
+        internal Task InitializeJobhostLanguageWorkerChannelAsync(int attemptCount)
+        {
+            // TODO: Add process managment for http invoker
+            _httpInvokerChannel = _httpInvokerChannelFactory.CreateHttpInvokerChannel(_scriptOptions.RootScriptPath, _metricsLogger, attemptCount);
+            _httpInvokerChannel.StartWorkerProcessAsync().ContinueWith(workerInitTask =>
+                 {
+                     if (workerInitTask.IsCompleted)
+                     {
+                         _logger.LogDebug("Adding http invoker channel. workerId:{id}", _httpInvokerChannel.Id);
+                         // TODO: wait for status api ok
+                         State = FunctionDispatcherState.Initialized;
+                     }
+                     else
+                     {
+                         _logger.LogWarning("Failed to start http invoker process. workerId:{id}", _httpInvokerChannel.Id);
+                     }
+                 });
+            return Task.CompletedTask;
+        }
+
+        public bool IsSupported(FunctionMetadata functionMetadata, string workerRuntime)
+        {
+            return true;
+        }
+
+        public async Task InitializeAsync(IEnumerable<FunctionMetadata> functions)
+        {
+            _functions = functions;
+
+            if (functions == null || functions.Count() == 0)
+            {
+                // do not initialize function dispachter if there are no functions
+                return;
+            }
+
+            State = FunctionDispatcherState.Initializing;
+            await InitializeJobhostLanguageWorkerChannelAsync(0);
+        }
+
+        public Task InvokeAsync(ScriptInvocationContext invocationContext)
+        {
+            return _httpInvokerService.GetInvocationResponse(invocationContext);
+        }
+
+        public async void WorkerError(HttpWorkerErrorEvent workerError)
+        {
+            if (!_disposing)
```

Comment:
Is this error handling logic copied from the OOP dispatcher?

Reply:
What can trigger worker errors? Will any invocation that fails to communicate with the worker end up here?

Reply:
this is the process exit error handler. if there is no  worker process that can handle invocation and retries to start the process fail, functions host is shutdown